File: backend/WebServiceApplication/src/main/java/com/foodforward/WebServiceApplication/model/dto/AuthDto.java
package com.foodforward.WebServiceApplication.model.dto;

public class AuthDto {
    private final String email;
    private final String password;
    private final String userId;

    public AuthDto(String email, String password, String googleIdToken, String userId) {
        this.email = email;
        this.password = password;
        this.userId = userId;
    }

    public AuthDto(String email, String password, String accessToken) {
        this.email = email;
        this.password = password;
        this.userId = accessToken;
    }

    public AuthDto(String email, String password) {
        this.email = email;
        this.password = password;
        this.userId = null;
    }

    public AuthDto(String accessToken) {
        this.userId = accessToken;
        this.email = null;
        this.password = null;
    }
    public AuthDto() {
        this.userId = null;
        this.email = null;
        this.password = null;
    }

    public String getEmail() {
        return email;
    }

    public String getPassword() {
        return password;
    }

    public String getUserId() {
        return userId;
    }
}

File: frontend/food_forward_app/lib/api/api-services/api-model/db-model/AuthDto.dart
class AuthDto {
  final String email;
  final String password;
  final String? userId;
  
  AuthDto({
    required this.email,
    required this.password,
    this.userId
  });


  // Convert a FileStorageRef object to a map
  Map<String, dynamic> toJson() {
    return {
      'email': email,
      'password': password,
      if (userId != null) 'userId': userId,
    };
  }
}
